Description: Remove bashism and switch to system's sh in init script
Author: Balint Reczey <rbalint@ubuntu.com>
Forwarded: no
Last-Update: 2017-10-27

--- a/etc/init.d/hibagent
+++ b/etc/init.d/hibagent
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/bin/sh
 # hibagent daemon
 # chkconfig: - 20 80
 # description: EC2 instance hibernation agent
@@ -16,7 +16,7 @@
 DAEMON=hibagent
 DAEMONOPTS="--config /etc/hibagent-config.cfg --pidfile $PIDFILE"
 
-if [[ $# -eq 0 ]]; then
+if [ $# -eq 0 ]; then
     echo "Usage: $0 {start|stop|restart}"
     exit 1
 else
@@ -26,7 +26,7 @@
 case $COMMAND in
 start)
 	printf "%-50s" "Starting $NAME..."
-	if [[ -f $PIDFILE ]]; then
+	if [ -f $PIDFILE ]; then
 		PID=`cat $PIDFILE`
 		if [[ "`ps axf | grep ${PID} | grep -v grep`" ]]; then
 			echo "hibagent is already running"
@@ -45,9 +45,9 @@
 ;;
 status)
 	printf "%-50s" "Checking $NAME..."
-	if [[ -f $PIDFILE ]]; then
+	if [ -f $PIDFILE ]; then
 		PID=`cat $PIDFILE`
-		if [[ -z "`ps axf | grep ${PID} | grep -v grep`" ]]; then
+		if [ -z "`ps axf | grep ${PID} | grep -v grep`" ]; then
 			printf "%s\n" "Process dead but pidfile exists"
 		else
 			echo "Running"
@@ -58,7 +58,7 @@
 ;;
 stop)
 	printf "%-50s" "Stopping $NAME"
-	if [[ -f $PIDFILE ]]; then
+	if [ -f $PIDFILE ]; then
 		PID=`cat $PIDFILE`
 		kill -HUP $PID
 		printf "%s\n" "Ok"
